port-chooser/src/test/java/org/terracotta/utilities/test/net/TestSupport.java

```diff
@@ -96,43 +140,62 @@ public static boolean sudoLsofWorks() {
             case 'p':
               currentProcess = portsPerProcess.get(line);
               if (currentProcess != null) {
-                LOGGER.warn("Unexpected output from {}: multiple '{}'", Arrays.toString(SUDO_LSOF_TCP), escape(line));
+                LOGGER.warn("Unexpected output from {}: multiple '{}'{}",
+                    Arrays.toString(lsofCommand), escape(line), collectEscapedLines(result));
                 return false;
               }
               currentProcess = new ArrayList<>();
               portsPerProcess.put(line, currentProcess);
               break;
+            case 'f':
+              if (currentProcess == null) {
+                LOGGER.warn("Unexpected output from {}; missing process '{}'{}",
+                    Arrays.toString(lsofCommand), escape(line), collectEscapedLines(result));
+                return false;
+              }
+              break;
             case 'n':
               if (currentProcess == null) {
-                LOGGER.warn("Unexpected output from {}; missing process '{}'", Arrays.toString(SUDO_LSOF_TCP), escape(line));
+                LOGGER.warn("Unexpected output from {}; missing process '{}'{}",
+                    Arrays.toString(lsofCommand), escape(line), collectEscapedLines(result));
                 return false;
               }
+              currentProcess.add(line);
               break;
             default:
-              LOGGER.warn("Unexpected output from {}: '{}'", Arrays.toString(SUDO_LSOF_TCP), escape(line));
+              LOGGER.warn("Unexpected output from {}: '{}'{}",
+                  Arrays.toString(lsofCommand), escape(line), collectEscapedLines(result));
               return false;
           }
         }
 
         if (portsPerProcess.isEmpty() || portsPerProcess.values().stream().allMatch(List::isEmpty)) {
-          String messages = "\n    " + String.join("\n    ", result);
-          LOGGER.debug("{} returned no ports:{}", Arrays.toString(SUDO_LSOF_TCP), messages);
+          LOGGER.debug("{} returned no ports:{}", Arrays.toString(lsofCommand), collectEscapedLines(result));
           return false;
         }
 
         return true;
 
       } else {
-        String messages = "\n    " + String.join("\n    ", result);
-        LOGGER.debug("Failed to run {}; rc={}{}", Arrays.toString(SUDO_LSOF_TCP), result.exitCode(), messages);
+        LOGGER.debug("Failed to run {}; rc={}{}", Arrays.toString(lsofCommand), result.exitCode(), collectEscapedLines(result));
         return false;
       }
     } catch (IOException e) {
-      LOGGER.debug("Failed to run {}", Arrays.toString(SUDO_LSOF_TCP), e);
+      LOGGER.debug("Failed to run {}", Arrays.toString(lsofCommand), e);
       return false;
     }
   }
 
+  private static String collectEscapedLines(Iterable<String> lines) {
+    String messages = StreamSupport.stream(lines.spliterator(), false)
+        .map(TestSupport::escape)
+        .collect(Collectors.joining("\n    "));
```

Comment:
Should this really be:
```java
    .map(s -> "    " + s)
    .collect(joining("\n"));
```

Reply:
Nah ... the concatenation _still_ instantiates a `StringBuilder` that, if you leave the full constant in `joining(...)`, isn't necessary.

Reply:
The former would get right of the need for the need to add a prefix to the non empty string. Also why do we need a  newline at the start?